src/baseline.rs

```diff
@@ -242,6 +243,52 @@ fn create_rustdoc_manifest_for_crate_version(
     }
 }
 
+fn choose_baseline_version(
+    crate_: &Crate,
+    version_current: Option<&semver::Version>,
+) -> anyhow::Result<String> {
+    // Try to avoid pre-releases
+    // - Breaking changes are allowed between them
+    // - Most likely the user cares about the last official release
+    if let Some(current) = version_current {
+        let mut instances = crate_
+            .versions()
+            .iter()
+            .filter_map(|i| {
+                semver::Version::parse(i.version())
+                    .map(|v| (v, i.is_yanked()))
+                    .ok()
+            })
+            // For unpublished changes when the user doesn't increment the version
+            // post-release, allow using the current version as a baseline.
+            .filter(|(v, _)| v <= current)
+            .collect::<Vec<_>>();
+        instances.sort();
+        instances
+            .iter()
+            .rev()
+            .find(|(v, yanked)| v.pre.is_empty() && !yanked)
+            .or_else(|| instances.last())
+            .map(|(v, _)| v.to_string())
+            .with_context(|| {
+                anyhow::format_err!(
+                    "No available baseline versions for {}@{}",
+                    crate_.name(),
+                    current
+                )
+            })
+    } else {
+        // If there is no normal version (not yanked and not a pre-release)
+        // choosing the latest one anyway is more reasonable than throwing an
+        // error, as there is still a chance that it is what the user expects.
```

Comment:
This comment isn't in the right place, and is confusing as a result. Its current placement seems to indicate that it's summarizing what's happening in the entire `else` block. But instead, the comment only applies to the `.unwrap_or_else(|| crate_.highest_version())` part.

I'd make that closure multi-line, and put the comment inside its `{` block instead. Then it's impossible to misinterpret.

Reply:
Definitely good idea, thank you!